src/eShop.ServiceDefaults/ConfigureSwaggerOptions.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Text;
+using Asp.Versioning.ApiExplorer;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Models;
+using Swashbuckle.AspNetCore.SwaggerGen;
+
+namespace eShop.ServiceDefaults;
+
+internal sealed class ConfigureSwaggerOptions : IConfigureOptions<SwaggerGenOptions>
+{
+    private readonly IApiVersionDescriptionProvider _provider;
+    private readonly IConfiguration _configuration;
+
+    public ConfigureSwaggerOptions(IApiVersionDescriptionProvider provider, IConfiguration configuration)
+    {
+        _provider = provider;
+        _configuration = configuration;
+    }
+
+    public void Configure(SwaggerGenOptions options)
+    {
+        foreach (var description in _provider.ApiVersionDescriptions)
+        {
+            options.SwaggerDoc(description.GroupName, CreateInfoForApiVersion(description));
+        }
+
+        ConfigureAuthorization(options);
+    }
+
+    private OpenApiInfo CreateInfoForApiVersion(ApiVersionDescription description)
+    {
+        /// {
+        ///   "OpenApi": {
+        ///     "Document": {
+        ///         "Title": ..
+        ///         "Version": ..
+        ///         "Description": ..
+        ///     }
+        ///   }
+        /// }
+        var openApi = _configuration.GetSection("OpenApi");
+        var document = openApi.GetRequiredSection("Document");
+        var info = new OpenApiInfo()
+        {
+            Title = document.GetRequiredValue("Title"),
+            Version = description.ApiVersion.ToString(),
+            Description = BuildDescription(description, document.GetRequiredValue("Description")),
+        };
+
+        return info;
+    }
+
+    private static string BuildDescription(ApiVersionDescription api, string description)
+    {
+        var text = new StringBuilder(description);
+
+        if (api.IsDeprecated)
+        {
+            if (text.Length > 0)
+            {
+                if (text[^1] == '.')
```

Comment:
Hey! Spotted this PR while searching on how to implement versioning. I believe this should be `!=`?
```suggestion
                if (text[^1] != '.')
```

Reply:
Good catch. Fixed.